File: frontend/src/types/customerMenu.ts
export interface MenuVariant {
  variant_id: number;
  variant_name: string;
  price: number;
}

export interface MenuItem {
  item_id: number;
  item_name: string;
  description: string | null;
  image_url: string | null;
  food_type: string;
  is_available: boolean;
  variants: MenuVariant[];
}

export interface Category {
  category_id: number;
  category_name: string;
  description: string | null;
  items: MenuItem[];
}

export interface CustomerMenuResponse {
  categories: Category[];
}

File: Backend/app/schemas/customer_menu.py
from decimal import Decimal
from pydantic import BaseModel


class MenuVariantResponse(BaseModel):
    variant_id: int
    variant_name: str
    price: Decimal

    class Config:
        from_attributes = True


class MenuItemResponse(BaseModel):
    item_id: int
    item_name: str
    description: str | None
    image_url: str | None
    food_type: str
    is_available: bool
    variants: list[MenuVariantResponse]

    class Config:
        from_attributes = True


class CategoryResponse(BaseModel):
    category_id: int
    category_name: str
    description: str | None
    items: list[MenuItemResponse]

    class Config:
        from_attributes = True


class CustomerMenuResponse(BaseModel):
    categories: list[CategoryResponse]